processing-status-notifications-fa/src/main/kotlin/gov/cdc/ocio/message/ReportParser.kt

```diff
@@ -0,0 +1,93 @@
+package gov.cdc.ocio.message
+
+import com.fasterxml.jackson.core.JsonFactory
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.ObjectMapper
+import gov.cdc.ocio.exceptions.ContentException
+import kotlin.collections.HashMap
+
+/**
+ * Class containing util methods to parse the content of the report
+ *
+ */
+class ReportParser {
+
+    private val reportTypeToStatusFieldMapping: MutableMap<String, String> = mutableMapOf("dex-hl7-validation" to "status",
+                                            "dex-file-copy" to "result",
+                                            "dex-metadata-verify" to "issues",
+                                            "upload" to "abc" // TODO : find this
+        )
+
+    /**
+     * Method which parses different types of notification reports for report status
+     * @param content String
+     * @return Unit
+     */
+    fun parseReportForStatus(content: String, reportType: String?): String {
+        val reportMetricCollector = HashMap<String, String>()
+        val factory = JsonFactory()
+        val mapper = ObjectMapper(factory)
+        val rootNode: JsonNode = mapper.readTree(content)
+
+        recurseParseHelper(rootNode, reportMetricCollector, reportType!!)
+        return reportMetricCollector[reportTypeToStatusFieldMapping[reportType]]!!
+    }
+
+    @Throws(ContentException::class)
+    private fun recurseParseHelper(node: JsonNode, reportMetricMap: HashMap<String, String>, reportType: String) {
```

Comment:
I think we're going to end up needing to define a report evaluation interface, which concrete implementations for each of the report types.  I just don't think we're going to be able to keep it generic like this.  For example, some reports we may have to look at more than one field to determine whether it was a success.  I think what you have is fine for now though.